cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -175,6 +175,81 @@ type_id to_type_id(SchemaElement const& schema,
   return type_id::EMPTY;
 }
 
+void metadata::sanitize_schema()
```

Comment:
neat approach, localizes the madness :+1:

Reply:
I think this could ultimately fix things like single-level list as well. I like this approach too.